src/hooks/useComments.ts

```diff
@@ -0,0 +1,21 @@
+import useSWR from 'swr';
+
+import { getComments, sendComment } from 'services/commentService';
+import { ISendComment } from '../interfaces';
+import { AxiosResponse } from 'axios';
+
+interface DataResponse<T> {
+  data: T | undefined;
+  isError: string;
+  sendComment: <T>(url: string, body: T) => Promise<AxiosResponse<ISendComment> | string>;
+}
+
+export function useComments<T>(url: string, key: string): DataResponse<T> {
```

Comment:
I don't quite understand why do we need this `key` prop? If you want to get access to the payload propery without knowing that property name you can use Object.keys(); `payload?.[Object.keys(payload)?.[0]]`